hermes_bridge/__main__.py

```
@@ -208,6 +226,45 @@ def main() -> None:
+                    proxy=_httpx.Proxy(_proxy_url_cap),
+                    transport=_httpx.HTTPTransport(socket_options=_sock_opts_cap),
+                )
+                return _orig_create(fresh, reason=reason, shared=shared)
+
+            agent._create_openai_client = _proxy_create
+            # Rebuild now so the initial client also gets the proxy.
+            agent.client = agent._create_openai_client(
+                agent._client_kwargs, reason="proxy_inject", shared=True
+            )
+            log.info("Patched _create_openai_client for proxy+keepalive (proxy=%s)", _proxy_url)
+        except Exception as _e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)